pydip/src/measurement.cpp

```diff
@@ -21,7 +21,11 @@
 #include "diplib/measurement.h"
 #include "diplib/chain_code.h"
 
-dip::MeasurementTool measurementTool;
+dip::MeasurementTool &measurementTool()
+{
+    static dip::MeasurementTool tool;
+    return tool;
+}
```

Comment:
> This fixes a really strange crash on shutdown in our `tst_DipImage.py` test, which seems to crash when trying to destruct the `featureIndices_` map on shutdown.

Interesting issue. Haven't seen in on macOS nor Linux so far. Oh, and Travis doesn't either on any of the three platforms. On the other hand, Travis doesn't test the use of any measurement functionality from within Python. Python tests are a bit sparse still.

I have no problem with this change though. There's also
```cpp
dip::ColorSpaceManager colorSpaceManager;
dip::Random randomNumberGenerator;
```
in `assorted.cpp`. I'm especially looking at `ColorSpaceManager`, which also keeps a map of names, similarly to `MeasurementTool`. But in `MeasurementTool` we changed the `std::map` to `tsl::robin_map`. Do you think that could be the culprit of your issues?

Reply:
this patch is relatively old - it definitely happened way before you added the tsl::robin_map. and it was only crashing within the part of our test suite which loaded the diplib code via a standalone python plugin (i.e. delayed dlopen) - so it may not be easy to reproduce. generally though, global statics are known to be evil and should be removed wherever possible. the pattern I show here is basically always superior, as it will defer the initialization until the first use, which is good for startup too.